src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/IKeyRewriterValueProvider.cs

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Mvc.ModelBinding
+{
+    /// <summary>
+    /// A value provider which can filter its contents to remove keys rewritten compared to the request data.
+    /// </summary>
+    public interface IKeyRewriterValueProvider : IValueProvider
```

Comment:
Do we want to consider making this an `internal` interface? The use cases for it don't look like something most users would ever need to query or consume.

Reply:
What argues in favour of an `internal` interface given (a) the method will be visible anyway in value providers that implement it and (b) we encourage users to write custom model binders i.e. this is another "if we need it, others might too" case?

Reply:
Visibility of the method isn't a good motivation to make the interface internal. This specific scenario is just very specific to how collection model binder + JQueryValueProvider works and doesn't seem very useful outside of that context. It'll be good to make the interface public with the right shape once we identify a use case outside of the collection binder.